### PR TITLE
[ADD] website_require_login.

### PR DESCRIPTION
**Pull request by Aldeigja**
This module allows to restrict access to specific website pages to logged users.

### CODE REVIEW

**Review by francesco-ooops — APPROVED** [review]
Functional ok!

File: website_require_login/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+{
+    "name": "Website Login Required",
+    "category": "Website",
+    "version": "14.0.1.0.0",
+    "author": "Advitus MB, Ooops, Odoo Community Association (OCA)",
+    "website": "https://github.com/OCA/website",
+    "license": "LGPL-3",
+    "depends": [
+        "website_sale",
```
Comment (GabbasovDinar) [inline website_require_login/__manifest__.py]: Why is there an depends on sale? 
Maybe it's enough to add only a website?

File: website_require_login/models/ir_http.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+class IrHttp(models.AbstractModel):
+    _inherit = "ir.http"
+
+    @classmethod
+    def _dispatch(cls):
+        res = super(IrHttp, cls)._dispatch()
+
+        # if not website request - skip
+        env = api.Environment(request.cr, SUPERUSER_ID, request.context)
```
Comment (GabbasovDinar) [inline website_require_login/models/ir_http.py]: Can we use `request.env` instead?

File: website_require_login/models/ir_http.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        website = env["website"].get_current_website()
+        if not website:
+            return res
+
+        auth_paths = (
```
Comment (GabbasovDinar) [inline website_require_login/models/ir_http.py]: it is better to search after checking `if request.uid == website.user_id.id`, so as not to search every time

**Review by GabbasovDinar — CHANGES_REQUESTED** [review]

File: website_require_login/__manifest__.py (hunk 1 above)
Comment (Aldeigja) [reply website_require_login/__manifest__.py]: Fixed

File: website_require_login/models/ir_http.py (hunk 3 above)
Comment (Aldeigja) [reply website_require_login/models/ir_http.py]: Fixed

File: website_require_login/models/ir_http.py (hunk 2 above)
Comment (Aldeigja) [reply website_require_login/models/ir_http.py]: Fixed

**Review by GabbasovDinar — APPROVED** [review]
Code LGTM

**Comment by francesco-ooops** [comment]
@OCA/website-maintainers what do you think?

**Comment by francesco-ooops** [comment]
@OCA/website-maintainers any chance for a review? :)

**Comment by francesco-ooops** [comment]
@tarteo @gfcapalbo could you take a look? thanks!

**Comment by francesco-ooops** [comment]
@OCA/website-maintainers trying again...

**Comment by francesco-ooops** [comment]
@rafaelbn can I request a review? :)

**Comment by rafaelbn** [comment]
I have to check, this module already exists I guess..

**Comment by rafaelbn** [comment]
> This module allows to restrict access to specific website pages to logged users.

This functionality is native in Odoo 15 for sure, is already in core

**Comment by rafaelbn** [comment]
OK, if Odoo 14 don't have this functionality is OK. As said v15 has it so this module is only for 14 😄 

The modules I said is for other controllers

- https://github.com/OCA/e-commerce/tree/15.0/website_sale_require_login
- https://github.com/OCA/event/tree/14.0/website_event_require_login

**Comment by francesco-ooops** [comment]
@rafaelbn good! This has indeed a more general approach to requiring login. 

Do you think it's good for merge?

**Comment by francesco-ooops** [comment]
@rafaelbn could you merge?

**Review by rafaelbn — APPROVED** [review]
/ocabot merge nobump

**Comment by francesco-ooops** [comment]
@rafaelbn and @OCA/website-maintainers any chance to make a PR to update copier? :)

**Comment by francesco-ooops** [comment]
@pedrobaeza can you launch merge command again?

**Pull request merged**